### PR TITLE
WFCORE-933 - argument validation starting embedded server

### PR DESCRIPTION
**Pull request by luck3y**

### CODE REVIEW

**Comment by luck3y** [comment]
retest this please

File: cli/src/main/java/org/jboss/as/cli/embedded/EmbedServerHandler.java [hunk 1]
```diff
@@ -176,6 +168,9 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
 
             List<String> cmdsList = new ArrayList<>();
             if (xml != null && xml.trim().length() > 0) {
+                if (!xml.endsWith(".xml")) {
+                    throw new CommandFormatException("The --server-config filename must end with .xml.");
+                }
```
Comment (bstansberry) [inline cli/src/main/java/org/jboss/as/cli/embedded/EmbedServerHandler.java]: This is fine as a workaround, but it smells funny that this is necessary. Can you file a JIRA to sort out why it's needed? Just a comment on the existing JIRA explaining why is fine if you know the answer and it's reasonable.

The check should be if (!xml.trim().endsWith(".xml")) or the trimming can just be done once and the result kept. :)

**Comment by bstansberry** [comment]
Other than the nit about trim(), looks good, but please squash the two commits.

File: cli/src/main/java/org/jboss/as/cli/embedded/EmbedServerHandler.java (hunk 1 above)
Comment (luck3y) [inline cli/src/main/java/org/jboss/as/cli/embedded/EmbedServerHandler.java]: Yep, I'd already opened https://issues.jboss.org/browse/WFCORE-991 for this. There's some suspicious code in ConfiguratioinFile.addSuffixToFile(historyBase, INITIAL) assuming that the rightmost . is part of a file suffix, and without it being present it appends the config name to the next . it encounters to create the path for the initial config file backup. This is almost certainly a directory that doesn't exist.

**Pull request merged**